src/main/javascript/crypto/src/tn-compat/Ticket.ts

```diff
@@ -0,0 +1,174 @@
+            signature = keys.signBytesWithEthereum(hexStringToArray(asn1Tic));
+        } catch (e) {
+            throw new Error(e);
+        }
+
+        me.createWithCommitment(devconId, ticketId, ticketClass, commitment, signature, keys);
+        return me;
+    }
+
+    private makeTicket() {
+        let ticket: string =
+            Asn1Der.encode('UTF8STRING', this.devconId)
+            + Asn1Der.encode('INTEGER', this.ticketId)
+            + Asn1Der.encode('INTEGER', this.ticketClass);
+        return Asn1Der.encode('SEQUENCE_30', ticket);
+    }
+
+    encodeSignedTicket(ticket: string)  {
+        let signedTicket:string =
+            ticket
+            + Asn1Der.encode('OCTET_STRING', uint8tohex(this.commitment))
+            + Asn1Der.encode('BIT_STRING', this.signature);
+        return Asn1Der.encode('SEQUENCE_30', signedTicket);
+    }
+
+    public getDerEncodingWithPK(): string {
+        let ticket = this.makeTicket();
+        let signedTicket: string =
+            ticket
+            + Asn1Der.encode('OCTET_STRING', uint8tohex(this.commitment))
+            + this.keys.getAsnDerPublic()
+            + Asn1Der.encode('BIT_STRING', this.signature);
+        return Asn1Der.encode('SEQUENCE_30', signedTicket);
+    }
+
+    public getDerEncoding():string {
+        return this.encoded;
+    }
+
+    public verify(): boolean {
+        return this.keys.verifyBytesWithEthereum(hexStringToArray(this.makeTicket()), this.signature);
+    }
+
+    public checkValidity(): boolean {
+        // The ticket is always valid on its own. It depends on which conference it is used
+        // and whether it has been revoked that decides if it can be used
+        return true;
+    }
+
+    public getTicketId():bigint {
+        return this.ticketId;
+    }
+
+    public getTicketClass(): number {
+        return this.ticketClass;
+    }
+
+    public getSignature(): string {
+        return this.signature;
+    }
+
+    static fromBase64(base64str: string, keys: KeyPair): Ticket {
+        let me = new this();
+        me.fromBytes(base64ToUint8array(base64str), keys);
+        return me;
+    }
+
+    fromBytes(bytes: Uint8Array, keys: KeyPair) {
+        const signedDevconTicket: SignedDevconTicket = AsnParser.parse(uint8toBuffer(bytes), SignedDevconTicket);
+
+        let devconId:string = signedDevconTicket.ticket.devconId;
+        let ticketId:bigint = BigInt(signedDevconTicket.ticket.ticketId);
+        let ticketClassInt:number = signedDevconTicket.ticket.ticketClass;
+
+        let commitment:Uint8Array = signedDevconTicket.commitment;
+        let signature:Uint8Array = signedDevconTicket.signatureValue;
+        this.createWithCommitment(devconId, ticketId, ticketClassInt, new Uint8Array(commitment), uint8tohex(new Uint8Array(signature)) , keys );
+    }
+
+    public getCommitment(): Uint8Array {
+        return this.commitment;
+    }
+
+
+    public getUrlEncoding() {
+        // TODO implement
```

Comment:
usually throw an error is good option, but in this case I prefer to use Ticket reader from main branch, so no need to update duplicated code. we can remove tn-compat/Ticket.ts because schema is not compatable with legacy ticket. main branch already can work with both ticket types.